code/modules/client/client_procs.dm

```diff
@@ -141,53 +141,31 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 /*
  * Call back proc that should be checked in all paths where a client can send messages
  *
- * Handles checking for duplicate messages and people sending messages too fast
+ * Handles checking for people sending messages too fast.
  *
- * The first checks are if you're sending too fast, this is defined as sending
- * SPAM_TRIGGER_AUTOMUTE messages in
- * 5 seconds, this will start supressing your messages,
- * if you send 2* that limit, you also get muted
+ * This is defined as sending SPAM_TRIGGER_AUTOMUTE (10) messages within 5 seconds of eachother, which gets you auto-muted.
+ *
+ * You will be warned if you send SPAM_TRIGGER_WARNING(5) messages withing 5 seconds of eachother to hopefully prevent false positives.
  *
- * The second checks for the same duplicate message too many times and mutes
- * you for it
  */
 /client/proc/handle_spam_prevention(message, mute_type)
+	if(!(CONFIG_GET(flag/automute_on)))
+		return FALSE
 
-	//Increment message count
-	total_message_count += 1
-
-	//store the total to act on even after a reset
-	var/cache = total_message_count
-
-	if(total_count_reset <= world.time)
-		total_message_count = 0
-		total_count_reset = world.time + (5 SECONDS)
-
-	//If they're really going crazy, mute them
-	if(cache >= SPAM_TRIGGER_AUTOMUTE * 2)
-		total_message_count = 0
-		total_count_reset = 0
-		cmd_admin_mute(src, mute_type, 1)
-		return 1
+	if(COOLDOWN_FINISHED(src, total_count_reset))
+		total_message_count = 0 //reset the count if it's been more than 5 seconds since the last message.
 
-	//Otherwise just supress the message
-	else if(cache >= SPAM_TRIGGER_AUTOMUTE)
-		return 1
+	total_message_count++
+	COOLDOWN_START(src, total_count_reset, 5 SECONDS)
```

Comment:
Won't this only clear if the person doesn't talk for 5 seconds? Would be really annoying hitting the chat filte limit because I didn't pause for 5 seconds between a conversation.

Reply:
Stop talking in choppy sentences. Put everything you want to say in one message instead of
talking
like this.

Reply:
And if you can type fast?

Reply:
Unless you need to make a sentence for 10 different people at once I don't really think its an issue. You can also type multiple sentences out in one say".

Reply:
Don't think this will be an issue to be honest